lightning/src/util/persist.rs

```diff
@@ -762,17 +762,18 @@ where
 
 	fn archive_persisted_channel(&self, funding_txo: OutPoint) {
 		let monitor_name = MonitorName::from(funding_txo);
-		let monitor = match self.read_monitor(&monitor_name) {
+		let monitor_key = monitor_name.as_str().to_string();
+		let monitor = match self.read_channel_monitor_with_updates(monitor_key) {
```

Comment:
FWIW, I'm not sure this bug is reachable at the moment because I think LDK will persist the full monitor on each block connection, and we don't archive until ~4k blocks have passed post-channel close. Still seems like a reasonable fix though in case things change.